doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,189 @@
+an `onMessage` function. In object form, a `shouldListen` function may also
+be optionally included to filter which channels to subscribe to by running
+the function on the name of the channel whenever a new channel is created.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+// Subscribe to a channel by name
+diagnostics_channel.subscribe('my-channel', (name, data) => {
+  // Received data
+});
+
+// Subscribe to any channels matching a function result
+diagnostics_channel.subscribe({
+  shouldListen(name) {
+    return /^\s+-channel/.test(name);
+  },
+  onMessage(name, data) {
+    // Received data
+  }
+});
+
+// Subscribe to every channel
+diagnostics_channel.subscribe((name, data) => {
+  // Received data
+});
+```
+
+#### `diagnostics_channel.publish(name, data)`
+
+* `name` {string} The channel name
+* `data` {Object} The message to send to the channel
+
+Lookup a channel by name and send a message, if it has active subscribers.
+It is recommended to use channel objects directly, in most cases, rather than
+using this top-level publish. This exists mainly for convenience in situations
+where publish performance is not important.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+diagnostics_channel.publish('my-channel', {
+  some: 'data'
+});
+```
+
+#### `diagnostics_channel.publisher(name)`
+
+* `name` {string} The channel name
+* Returns: {Publisher} The named channel object
+
+This is the primary entry-point to anyone wanting to publish data to a named
+channel. It produces a channel object which is optimized to reduce overhead at
+publish time as much as possible.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.publisher('my-channel');
+```
+
+### Class: `Publisher`
+
+The class `Publisher` represents an individual named channel within the data
+pipeline. It is primarily used by the publishing side of the event system to
+check if there are subscribers and to send messages.
+
+#### `publisher.shouldListen()`
+
+* Returns: {boolean} If there are active subscribers
+
+Check if there are active subscribers to this channel.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const publisher = diagnostics_channel.publisher('my-channel');
+
+if (publisher.shouldListen()) {
+  // There are subscribers
+}
+```
+
+#### `publisher.ifActive`
+
+* Returns: {Publisher} Returns the publisher when active
+
+The `ifActive` property is an alternative to `shouldPublish()` which is
+intended for use with optional-chaining. This allows the following call
+to `publish()` to be skipped along with the creation of its input object.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const publisher = diagnostics_channel.publisher('my-channel');
+
+publisher.ifActive?.publish({
+  some: 'data'
+});
+```
+
+#### `publisher.publish(data)`
+
+* `data` {Object} The message to send to the channel subscribers
```

Comment:
are we really limited to `Object` here? Maybe someone likes to publish also strings or numbers?

Reply:
It can be anything. I'll update that soon. :)